velox/type/Timestamp.h

```diff
@@ -115,31 +115,27 @@ struct Timestamp {
   void toTimezone(int16_t tzID);
 
   bool operator==(const Timestamp& b) const {
-    return seconds_ == b.seconds_ && nanos_ == b.nanos_;
+    return toNanos() == b.toNanos();
   }
 
   bool operator!=(const Timestamp& b) const {
-    return seconds_ != b.seconds_ || nanos_ != b.nanos_;
+    return toNanos() != b.toNanos();
   }
 
   bool operator<(const Timestamp& b) const {
-    return seconds_ < b.seconds_ ||
-        (seconds_ == b.seconds_ && nanos_ < b.nanos_);
+    return toNanos() < b.toNanos();
```

Comment:
The previous comparison cannot ensure `a is less than b`, found with unit test.

Reply:
In which case?  Can you explain?

Reply:
The case is `seconds_` is less than `b.seconds_` but `nanos_` is much larger than `b.nanos_` and compensates the gap of `seconds`. Do we consider rounding the nanos to seconds as much as possible in Timestamp constructor, so previous logic can still work?

Reply:
Yes we should make sure `nanos_` is less than 1e9